lib/net.js

```diff
@@ -381,14 +387,36 @@ Socket.prototype.read = function(n) {
 };
 
 Socket.prototype.setTimeout = function(msecs, callback) {
+  // Type checking identical to timers.enroll()
```

Comment:
This might be enough code to factor it out into some internal function, wdyt?

Reply:
Might be a good idea yeah.